store/storetest/post_store.go

```diff
@@ -1783,3 +1783,8 @@ func testPostStoreGetOldest(t *testing.T, ss store.Store) {
 
 	assert.EqualValues(t, o2.Id, r1.Id)
 }
+
+func TestMaxPostSize(t *testing.T, ss store.Store) {
```

Comment:
This won't be found automatically. You need to add it to the `TestPostStore` function at the top of this file.

This is a pitfall I was hoping to remedy by integrating dead code checks into our CI via [dominikh/go-tools/tree/master/cmd/unused](https://github.com/dominikh/go-tools/tree/master/cmd/unused). (Just made a JIRA issue [here](https://mattermost.atlassian.net/browse/MM-9910).)

Reply:
Ah, good catch! I decided to relocate the unit tests for the business logic into the store, but then clearly didn't prove that the tests were still running.

Separately, this structuring has always confused me. I get that it ensures we run the tests for each type of store, but it feels like it would be easier to grok if each test stood alone and instead reached out to get the list of stores over which to iterate. I will add this to a list of things I want to revisit.

Reply:
My thinking was that I wanted testing stores by package to work (`go test ./store/sqlstore`, `go test ./store/nosqlstore`, etc.), but maybe it's not worth the trade-off. ¯\\\_(ツ)\_/¯